### PR TITLE
Ability to switch to different VM providers by config file (OS X only)

### PR DESCRIPTION
**Pull request by taiansu**
This is for #107

  1. Read the config from `$HOME/.docker/vm.config`. Looking for `PROVIDER="Parallels Desktop"` or `PROVIDER="VMware Fusion"` then setup the command accordingly. 
  2. Only test the Parallels Desktops for OS X && PD 11. __Please test with VMWare before merge.__

Thanks!

### CODE REVIEW

File: osx/mpkg/quickstart.app/Contents/Resources/Scripts/start.sh [hunk 1]
```diff
@@ -1,8 +1,26 @@
 #!/bin/bash
 
+CONFIG_FILE="$HOME/.docker/vm.config"
+[[ -f $CONFIG_FILE ]] && source $CONFIG_FILE
+
+if [ "$PROVIDER" == "VMWare Fusion" ]; then
+  PROVIDER_CLI=/usr/local/bin/vmrun
+  STATUS_CMD="list | grep"
```
Comment (taiansu) [inline osx/mpkg/quickstart.app/Contents/Resources/Scripts/start.sh]: Wondering if this works or not. :sweat:

**Comment by jeffdm** [comment]
Hi there, thanks for the PR, and sorry for the delay in getting back. Taking a look, though I feel this may be a large change due to both Kitematic and the Docker Quickstart Terminal depending on VBox.

@nathanleclaire what do you think?

**Comment by taiansu** [comment]
No problem at all, @jeffdm. I tried to find how to do fix the Kitematic but failed. If anyone have time to point out which file should I look into, I might able to help.

 IMHO we should only merge this feature once we have all OSX things workable. But I'll leave the decision to you on how to deal with this PR. :laughing:

**Comment by nathanleclaire** [comment]
Hi @taiansu, thanks for the PR.  My feeling on this is that it's not the right fit to put directly inside of the Toolbox.  Toolbox is intended to provide an opinionated workflow to users to get started as quickly as possible and one of those opinions is that VirtualBox should be used for the provider (it's not amazing but, it is free and widely available).

I wonder if some work that @FrenchBen has been doing lately on using machines other than `default` with Kitematic might help with this use case though?  In general being able to switch the "active" machine from the Toolbox somehow seems like a decent idea.

**Comment by taiansu** [comment]
Hi @jeffdm @nathanleclaire 
Sorry for late reply, I just move to a new country with unstable network connection recently.

And thanks for the explanation. I understand that this pr might not fit the need of original issue. You can close this pr, no hard feelings. :smile: 

But just wondering how do you think if I remove all the actual provider things, just leave the hook to read commands from the config, and keep the behavior unchanged if no config found? That way I don't have to manually patch the file along the updates. :bulb:

**Pull request closed without merging**